scanamo/src/main/scala/org/scanamo/DynamoFormat.scala

```diff
@@ -56,8 +56,6 @@ trait DynamoFormat[T] {
 
   def iso[U](r: T => U, w: U => T): DynamoFormat[U] = DynamoFormat.iso(r, w)(this)
   def xmap[U](r: T => Either[DynamoReadError, U], w: U => T): DynamoFormat[U] = DynamoFormat.xmap(r, w)(this)
-  def coercedXmap[U](read: T => U, write: U => T): DynamoFormat[U] =
```

Comment:
Yeah, the type is a constraint on [`catchOnly`](https://github.com/typelevel/cats/blob/e0784fc0d95b9c4eed0bb1ddce6c69e3fd9a4a4f/core/src/main/scala/cats/syntax/either.scala#L365). I'm not so sure removing is the right way to go, as people may be using it. I actually don't like the companion methods, it is way more idiomatic to write:
```scala
DynamoFormat[Int].coercedXmap(_.toString, _.toInt)
```
True, scanamo does not have good story for manually-written instances, but it could with a improvements.

This should please the scala 3 compiler
```
DynamoFormat.coercedXmap(read, write)(this, implicitly[ClassTag[Throwable]])
```

After all, `Throwable >: Null <: Throwable` and functions are by nature impure in Scala. In Scala 3 we may be able to change the type to:

```scala
def coercedXmap[Ex >: Null <: Throwable: ClassTag]
  ( read: T => U canThrow Ex
  , write: U => T canThrow Ex
  ): DynamoFormat[U]
```
... [one day perhaps](https://github.com/lampepfl/dotty/blob/b65958f139a0a818d36ee04468ada93b8fb472ac/docs/docs/reference/experimental/canthrow.md).

Reply:
> I'm not so sure removing is the right way to go, as people may be using it.

This is true, it's hard to know. I did just now do some searches on GitHub, to see if I could find any examples of it's use. I checked:

* The 40 occurrences of `coercedXmap` in [the public & private repos of the Guardian](https://github.com/search?q=org%3Aguardian+coercedXmap&type=Code)
* The 31 occurrences of `coercedXmap` in [public non-Guardian repos](https://github.com/search?q=coercedXmap+-org%3Aguardian&type=Code)

...**every single one of those occurrences was using the static `DynamoFormat.coercedXmap` form** - none of them using the instance method. Of course, there may well be lots of code in private repositories that I can't see that _is_ using the instance version!

> This should please the scala 3 compiler
> 
> ```
> DynamoFormat.coercedXmap(read, write)(this, implicitly[ClassTag[Throwable]])
> ```

I've given this a try out, and it does indeed please the Scala 3 compiler, thank you! However, although I agree that the idiomatic form of `DynamoFormat[Int].coercedXmap(_.toString, _.toInt)` is potentially nice, the fact that it seems currently unused, and perhaps catches a lot more exceptions than users would expect (eg `java.lang.OutOfMemoryError`) I think a better solution probably _would_ be to remove the method.

My focus for the moment is getting Scala 3 working though, so if you feel it's better to do the `implicitly` fix, I'll raise a new PR for that.